### PR TITLE
MAINT: add test_file solves #1026

### PR DESCRIPTION
**Pull request by adler-j**
Finally getting this one out of the way.

### CODE REVIEW

File: odl/util/testutils.py [hunk 1]
```diff
@@ -700,5 +700,22 @@ def run_doctests(skip_if=False, **kwargs):
     testmod(optionflags=optionflags, extraglobs=extraglobs, **kwargs)
 
 
+def test_file(file, args=None):
+    """Run tests in file with proper default arguments."""
+    try:
+        import pytest
+    except ImportError:
+        raise ImportError('ODL tests cannot be run without `pytest` installed.'
+                          '\nRun `$ pip install [--user] odl[testing]` in '
```
Comment (kohr-h) [inline odl/util/testutils.py]: Sure we want to always recommend the `pip install` option? People will blindly follow instructions and potentially mess up their environments. I would rather add "or `conda install pytest`" to the message.

And, why so complicated? This utility just needs plain `pytest`, no other plugins.

Comment (adler-j) [reply odl/util/testutils.py]: Well pup is always present, competent users will know what to do in other cases.

W.r.t complications, there is simply some stuff we need to do, perhaps I should document them.

Specifically we do a fix to string formatting for Windows, add verbose output and ensure stdout is correctly captured.

Comment (kohr-h) [reply odl/util/testutils.py]: > Well pup is always present, competent users will know what to do in other cases.

Beginners in the field increasingly use conda as go-to solution for virtual environments. I feel we should help them by not favoring the slightly wrong solution for that scenario.



> W.r.t complications, there is simply some stuff we need to do, perhaps I should document them.
> 
> Specifically we do a fix to string formatting for Windows, add verbose output and ensure stdout is correctly captured.

That's not what I meant. It was about the `$ pip install [--user] odl[testing]` thing, which installs also `pytest-pep8` and `pytest-cov` which are not needed. So a simple `pip install pytest` will do.

Comment (adler-j) [reply odl/util/testutils.py]: > Beginners in the field increasingly use conda as go-to solution for virtual environments. I feel we should help them by not favoring the slightly wrong solution for that scenario.

Well what issues could they expect from that? I've never noticed any at all myself (except for stuff that is compiled ofc)

Comment (kohr-h) [reply odl/util/testutils.py]: The most significant one is that pip packages won't be updated unless explicitly updated with `pip -U`. But who does that?
And, if a package is already installed by conda and then a user runs `pip install` with the same package, pip will overwrite the old one but not delete the conda package list entry. Then you have to force-uninstall them later.

So it's not a major thing, but can be an annoyance.

Comment (adler-j) [reply odl/util/testutils.py]: Frankly, I feel that stuff like this (while important) should be considered more a "bug" with the ecosystem, and not something we should explicitly handle unless it really creates lots of issues.

My proposal is that we always propose pip unless the packages are compiled, in which case conda should at least be mentioned.

Comment (kohr-h) [reply odl/util/testutils.py]: Well, alternatively we could simply say "pleas install package X", not mentioning any tool. At least in those error messages, where we cannot be super-detailed. The docs can clarify.

**Pull request merged**